src/main/java/org/embulk/output/sftp/SftpUtils.java

```diff
@@ -0,0 +1,279 @@
+    {
+        FileSystemOptions fsOptions = new FileSystemOptions();
+
+        try {
+            SftpFileSystemConfigBuilder builder = SftpFileSystemConfigBuilder.getInstance();
+            builder.setUserDirIsRoot(fsOptions, task.getUserDirIsRoot());
+            builder.setTimeout(fsOptions, task.getSftpConnectionTimeout() * 1000);
+            builder.setStrictHostKeyChecking(fsOptions, "no");
+            if (task.getSecretKeyFilePath().isPresent()) {
+                IdentityInfo identityInfo = new IdentityInfo(
+                    new File((task.getSecretKeyFilePath().transform(localFileToPathString()).get())),
+                    task.getSecretKeyPassphrase().getBytes()
+                );
+                builder.setIdentityInfo(fsOptions, identityInfo);
+                logger.info("set identity: {}", task.getSecretKeyFilePath().get());
+            }
+
+            if (task.getProxy().isPresent()) {
+                ProxyTask proxy = task.getProxy().get();
+
+                ProxyTask.ProxyType.setProxyType(builder, fsOptions, proxy.getType());
+
+                if (proxy.getHost().isPresent()) {
+                    builder.setProxyHost(fsOptions, proxy.getHost().get());
+                    builder.setProxyPort(fsOptions, proxy.getPort());
+                }
+
+                if (proxy.getUser().isPresent()) {
+                    builder.setProxyUser(fsOptions, proxy.getUser().get());
+                }
+
+                if (proxy.getPassword().isPresent()) {
+                    builder.setProxyPassword(fsOptions, proxy.getPassword().get());
+                }
+
+                if (proxy.getCommand().isPresent()) {
+                    builder.setProxyCommand(fsOptions, proxy.getCommand().get());
+                }
+            }
+        }
+        catch (FileSystemException e) {
+            logger.error(e.getMessage());
+            throw new ConfigException(e);
+        }
+
+        return fsOptions;
+    }
+
+    SftpUtils(PluginTask task)
+    {
+        this.manager = initializeStandardFileSystemManager();
+        this.userInfo = initializeUserInfo(task);
+        this.fsOptions = initializeFsOptions(task);
+        this.host = task.getHost();
+        this.port = task.getPort();
+        this.maxConnectionRetry = task.getMaxConnectionRetry();
+    }
+
+    public void close()
+    {
+        manager.close();
+    }
+
+    public Void uploadFile(final File localTempFile, final String remotePath)
+    {
+        try {
+            return retryExecutor()
+                    .withRetryLimit(maxConnectionRetry)
+                    .withInitialRetryWait(500)
+                    .withMaxRetryWait(30 * 1000)
+                    .runInterruptible(new Retryable<Void>() {
+                        @Override
+                        public Void call() throws IOException
+                        {
+                            FileObject remoteFile = newSftpFile(getSftpFileUri(remotePath));
```

Comment:
Minor (optional): we can have multiple resources in one `try (...)`:
```java
try (FileObject remoteFile = newSftpFile(getSftpFileUri(remotePath));
     BufferedOutputStream outputStream = new BufferedOutputStream(remoteFile.getContent().getOutputStream());
     BufferedInputStream inputStream = new BufferedInputStream(new FileInputStream(localTempFile))
     ) {
    logger.info("new sftp file: {}", remoteFile.getPublicURIString());
    IOUtils.copy(inputStream, outputStream);
}
```
It can help to ensure all resources are closed properly :)